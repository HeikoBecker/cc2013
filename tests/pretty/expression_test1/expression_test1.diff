1,22c1,82
< int (a(char c))
< {
< 	((!b) == f);
< 	((a.b)->c);
< 	((((a[7])[10])[c]).c);
< 	(bcud != 5);
< 	(k = ((a && b) || c));
< 	((a == b) != c);
< 	((a != b) == c);
< 	((a < b) == c);
< 	(a == (b < c));
< 	((a < (!b)) < (&x));
< 	(a());
< 	((a[i])());
< 	(((a[i]).j)->p);
< 	(a[(a == b)]);
< 	(a[(a == b)]);
< 	"bla";
< 	c;
< 	8;
< 	(a[8]);
< }
---
>  DECL : a : function (char) returning int
>  DECL : c : char
> [1;33mLexer: [0mAdvancing... got i
> [1;33mLexer: [0mAdvancing... got n
> [1;33mLexer: [0mAdvancing... got t
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mAdvancing... got a
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got a
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mAdvancing... got (
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got (
> [1;33mLexer: [0mAdvancing... got c
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got c
> [1;33mLexer: [0mAdvancing... got h
> [1;33mLexer: [0mAdvancing... got a
> [1;33mLexer: [0mAdvancing... got r
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mAdvancing... got c
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got c
> [1;33mLexer: [0mAdvancing... got )
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got )
> [1;33mLexer: [0mAdvancing... got {
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got {
> [1;33mLexer: [0mAdvancing... got 
> 
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got 
> 
> [1;33mLexer: [0mAdvancing... got !
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got !
> [1;33mLexer: [0mAdvancing... got b
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got b
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mAdvancing... got =
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got =
> [1;33mLexer: [0mAdvancing... got =
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got  
> [1;33mLexer: [0mAdvancing... got f
> [1;33mLexer: [0mrewinding...
> 
> [1;33mLexer: [0mAdvancing... got f
> [1;33mLexer: [0mAdvancing... got ;
> [1;33mLexer: [0mrewinding...
> 
> expression_test1.c:2:2: error: b is not declared in this scope
> !b == f;
>  ^^^^
> 1 error(s)
