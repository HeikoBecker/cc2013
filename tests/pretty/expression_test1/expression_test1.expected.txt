struct C
{
	int c;
	int p;
};

struct S
{
	struct C (*b);
	int c;
	struct C (*j);
};

int (a(int c))
{
	struct S a;
	int b;
	int f;
	struct S (*(*(*d)));
	struct S (*da);
	int bcud;
	int k;
	int i;
	((!b) == f);
	((a.b)->c);
	((((d[7])[10])[c]).c);
	(bcud != 5);
	(k = ((b && b) || c));
	((f == b) != c);
	((f != b) == c);
	((f < b) == c);
	(f == (b < c));
	int x;
	((f < (!b)) < (&x));
	int ((*g)(void));
	(g());
	(((da[i]).j)->p);
	(d[(f == b)]);
	(d[(f == b)]);
	"bla";
	c;
	8;
	(d[8]);
}
